### PR TITLE
Add integration with DeepL translation services

### PR DESCRIPTION
**Pull request by airibarne**
## Summary
We propose a new integration of [DeepL](https://www.deepl.com/) translation services into the Haystack framework. 

This integration allows users to utilize DeepL's powerful translation capabilities directly within Haystack, 
enhancing the multilingual processing of documents and queries. The integration relies on the official [DeepL Python client](https://github.com/DeepLcom/deepl-python) to interact with the [translation service API](https://www.deepl.com/en/pro-api)

## Motivation
This integration addresses the need for high-quality translations in multilingual applications. DeepL is known for its superior translation accuracy, which will significantly benefit Haystack users working with non-English languages, enhancing both the user experience and the accuracy of information retrieval.

## Questions
The main doubt here is the _Translator_ appellation for the component, since such concept does not exist -yet- in Haystack. 

While these new components could fit into existing component kinds, such as _Generator_ or even _PreProcessor_, we think translation services could be themselves a new _Translator_ kind, in which, at some point, there would be other providers components such as Google Translate.

### CODE REVIEW

File: integrations/deepl.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+[DeepL](https://www.deepl.com/) is a powerful translation services provider, offering high-quality translations 
+in multiple languages. This integration allows you to use DeepL's translation services with Haystack. 
+
+## Installation
+
+```console
+pip install deepl-haystack
+```
+
+## Usage
```
Comment (TuanaCelik) [inline integrations/deepl.md:39]: Here or below, could you mention that the user should export the `DEEPL_API_KEY` environment variable? We are trying to recommend environment variables as the way to deal with api keys 🙏 
That way, you could remove the `.from_token` in the initialization code below

**Comment by TuanaCelik** [comment]
In terms of the naming "Translator" - the component names, having `DeepL` as the first word make sense to me. Pre Haystack 2.0 we also had components called translators, and I can imagine that this concepts comes back to Haystack one day. In any case, calling these `DeepLDocumentTranslator` etc makes sense to me.

**Comment by julian-risch** [comment]
`DeepLDocumentTranslator` and `DeepLTextTranslator` as component names make sense to me too. 👍

File: integrations/deepl.md (hunk 1 above)
Comment (airibarne) [reply integrations/deepl.md:39]: Just pushed a new version of the file without `Secret` usage while suggesting env var usage, which I agree is a better practice :+1:

**Review by TuanaCelik — APPROVED** [review]
LGTM, we're ready to merge 🤝

**Comment by airibarne** [comment]
Great, thank you!! Good to see it out there that soon, congrats on the great work you're doing here with Haystack :smile:

Cheers :raised_hands:

**Pull request merged**